cmd/kubeadm/app/util/runtime/runtime.go

```diff
@@ -0,0 +1,174 @@
+	var toolName string
+	var cruntime ContainerRuntime
+
+	if criSocket != kubeadmapiv1alpha2.DefaultCRISocket {
+		toolName = "crictl"
+		// !!! temporary work around crictl warning:
+		// Using "/var/run/crio/crio.sock" as endpoint is deprecated,
+		// please consider using full url format "unix:///var/run/crio/crio.sock"
+		if filepath.IsAbs(criSocket) && runtime.GOOS != "windows" {
+			criSocket = "unix://" + criSocket
+		}
+		cruntime = &CRIRuntime{execer, criSocket}
+	} else {
+		toolName = "docker"
+		cruntime = &DockerRuntime{execer}
+	}
+
+	if _, err := execer.LookPath(toolName); err != nil {
+		return nil, fmt.Errorf("%s is required for container runtime: %v", toolName, err)
+	}
+
+	return cruntime, nil
+}
+
+// IsDocker returns true if the runtime is docker
+func (cruntime *CRIRuntime) IsDocker() bool {
+	return false
+}
+
+// IsDocker returns true if the runtime is docker
+func (cruntime *DockerRuntime) IsDocker() bool {
+	return true
+}
+
+// IsRunning checks if runtime is running
+func (cruntime *CRIRuntime) IsRunning() error {
+	if err := cruntime.exec.Command("crictl", "-r", cruntime.criSocket, "info").Run(); err != nil {
+		return fmt.Errorf("container runtime is not running: %v", err)
+	}
+	return nil
+}
+
+// IsRunning checks if runtime is running
+func (cruntime *DockerRuntime) IsRunning() error {
+	if err := cruntime.exec.Command("docker", "info").Run(); err != nil {
+		return fmt.Errorf("container runtime is not running: %v", err)
+	}
+	return nil
+}
+
+// ListKubeContainers lists running k8s CRI pods
+func (cruntime *CRIRuntime) ListKubeContainers() ([]string, error) {
+	output, err := cruntime.exec.Command("crictl", "-r", cruntime.criSocket, "pods", "-q").CombinedOutput()
+	if err != nil {
+		return nil, err
+	}
+	pods := []string{}
+	for _, pod := range strings.Fields(string(output)) {
+		if strings.HasPrefix(pod, "k8s_") {
+			pods = append(pods, pod)
+		}
+	}
+	return pods, nil
+}
+
+// ListKubeContainers lists running k8s containers
+func (cruntime *DockerRuntime) ListKubeContainers() ([]string, error) {
+	output, err := cruntime.exec.Command("docker", "ps", "-a", "--filter", "name=k8s_", "-q").CombinedOutput()
+	return strings.Fields(string(output)), err
+}
+
+// RemoveContainers removes running k8s pods
+func (cruntime *CRIRuntime) RemoveContainers(containers []string) []error {
+	errs := []error{}
+	for _, container := range containers {
+		err := cruntime.exec.Command("crictl", "-r", cruntime.criSocket, "stopp", container).Run()
+		if err != nil {
+			// don't stop on errors, try to remove as many containers as possible
+			errs = append(errs, fmt.Errorf("failed to stop running pod %s: %v", container, err))
+		} else {
+			err = cruntime.exec.Command("crictl", "-r", cruntime.criSocket, "rmp", container).Run()
+			if err != nil {
+				errs = append(errs, fmt.Errorf("failed to remove running container %s: %v", container, err))
+			}
+		}
+	}
+	return errs
```

Comment:
`return errors.NewAggregate(errs)` where errors is `"k8s.io/apimachinery/pkg/util/errors"`